### PR TITLE
Add documentation for stock items and stock movements

### PR DESCRIPTION
**Pull request by benjaminwil**
This pull request adds articles about `Spree::StockItem`s and `Spree::StockMovement`s. (Stock locations documentation is summarized in #2538.)

_This is part a larger project to improve Solidus's documentation. [See this gist with the high-level table of contents](https://gist.github.com/benjaminwil/5e2d2da3d9def6b949b833b1aa2549d3). Where and how this documentation will exist is still up for discussion._

### CODE REVIEW

File: guides/inventory/stock-items.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Stock items
+
+On-hand inventory is tracked using the `Spree::StockItem` model. Each
+`Spree::Variant` in a store has a corresponding `Spree::StockItem` object with a
+`count_on_hand` value that represents the number of items you have in stock.
```
Comment (kennyadsl) [inline guides/inventory/stock-items.md]: > Each `Spree::Variant` in a store has a corresponding `Spree::StockItem` object

it has a corresponding `Spree::StockItem` object for each stock location present in the store. So the same variant can have more than one `Spree::StockItem` associated objects.

File: guides/inventory/stock-items.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+### Changing the count on hand value
+
+While a `Spree::StockMovement` object logs the increase or decrease of the
+`count_on_hand` value, administrators can also edit the count on hand from the
+`solidus_backend`.
+
+Whenever an administrator updates the count on hand, they are discarding the old
+value completely. So, if a stock item is
+[backorderable](#backorderable-stock-items) and its `count_on_hand` is `-5`,
```
Comment (kennyadsl) [inline guides/inventory/stock-items.md]: for the sake of the example, is it relevant to say that the item should be backorderable? Maybe it's more clear if we remove this concept from here and use a positive number for the initial `count_on_hand`. WDYT?

File: guides/inventory/stock-items.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+See the [Stock movements][stock-movements] article for more information.
+
+[stock-movements]: stock-movements.md
+
+## Backorderable stock items
+
+If a `Spree::StockItem` is `backorderable`, then customers can continue to order
+it after the product is sold out. When a sold out product continues to sell, the
+`count_on_hand` would become a negative integer.
```
Comment (kennyadsl) [inline guides/inventory/stock-items.md]: Maybe here we should say that orders that contains backorderable stock item/inventory units (not sure which concept it's better to mention here in order to not increase complexity) can not be shipped until that stock item has been restocked in the stock location. Not sure if we have other guide for this topic, maybe we can just link there?

Comment (benjaminwil) [reply guides/inventory/stock-items.md]: There aren't any other guides on this topic yet, so I'm going to add a bit more information here. Thanks for pointing this out!

**Comment by benjaminwil** [comment]
@kennyadsl Thanks for the review. I've addressed all your comments. :+1:

**Pull request merged**